06-express-and-layered-architecture/src/cart/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { users } from "../../models/user.entity";
+
+export async function checkIfUserExists(req: any, res: any, next: any) {
+  const userId = req.header("x-user-id");
+
+  if (!userId) {
+    return res.status(400).send("User ID is not provided");
+  }
+
+  const user = users.find((u) => u.id === userId);
+
+  if (!user) {
+    return res.status(404).send("User not found");
```

Comment:
Good job for sending the correct status code here as well! 404 is the way to go if a resource is not found!